external/kafka/src/main/scala/org/apache/spark/streaming/kafka/NewDirectKafkaInputDStream.scala

```diff
@@ -0,0 +1,134 @@
+/*
```

Comment:
Back to Hari's comment about the word 'new'. On the JIRA we discussed and ruled out using version numbers. It might have been easier if like 'HighLevelAPI/LowLevelAPI/SimpleConsumer', there was a word with a functional element to it that we could use here for the 'newConsumer'